numbers/src/test/java/org/dicio/numbers/lang/fr/FrenchFormatter.java

```diff
@@ -0,0 +1,236 @@
+package org.dicio.numbers.lang.fr;
+
+import java.time.LocalTime;
+
+import org.dicio.numbers.formatter.NumberFormatter;
+import org.dicio.numbers.util.MixedFraction;
+
+public class FrenchFormatter extends NumberFormatter {
+
+    protected FrenchFormatter(String configFolder) {
+        super("config/fr-fr");
+    }
+
+    // Copied from EnglishFormatter with no changes except for strings.
+    @Override
+    public String niceNumber(MixedFraction mixedFraction, boolean speech) {
+        if (speech) {
+            final String sign = mixedFraction.negative ? "moins " : "";
+            if (mixedFraction.numerator == 0) {
+                return sign + pronounceNumber(mixedFraction.whole, 0, true, false, false);
+            }
+
+            String denominatorString;
+            if (mixedFraction.denominator == 2) {
+                denominatorString = "moitié";
+            } else if (mixedFraction.denominator == 4) {
+                denominatorString = "quart";
+            } else {
+                // use ordinal: only half and quarter are exceptions
+                denominatorString
+                        = pronounceNumber(mixedFraction.denominator, 0, true, false, true);
+            }
+
+            final String numeratorString;
+            if (mixedFraction.numerator == 1) {
+                numeratorString = "un";
+            } else {
+                numeratorString = pronounceNumber(mixedFraction.numerator, 0, true, false, false);
+                denominatorString += "s";
+            }
+
+            if (mixedFraction.whole == 0) {
+                return sign + numeratorString + " " + denominatorString;
+            } else {
+                return sign + pronounceNumber(mixedFraction.whole, 0, true, false, false)
+                        + " et " + numeratorString + " " + denominatorString;
+            }
+
+        } else {
+            return niceNumberNotSpeech(mixedFraction);
+        }        return null;
```

Comment:
Was this `return null` left behind by accident?
```suggestion
        }
```

Reply:
Yes, again got too used to IDEA, my bad. I'll switch back as to avoid further nonsense.